### PR TITLE
[#1553] Ignore blank cells at the end of header row when importing reports

### PR DESCRIPTION
**Pull request by muloem**

### CODE REVIEW

File: GAE/src/org/waterforpeople/mapping/dataexport/RawDataSpreadsheetImporter.java [hunk 1]
```diff
@@ -758,6 +767,26 @@ protected void invokeUrl(String serverBase, String urlString,
...
+    /**
+     * When a blank or null cell is incurred while processing, make sure that this is the last cell
+     * in the row and ignore any other "phantom cells" that may occur. We only allow this for the
+     * header row. If the blank cell occurs in between valid header cells we return true.
+     *
+     * @param cell
+     * @return
+     */
+    private boolean isMissingHeaderCell(Cell cell) {
+        assert cell.getRow().getRowNum() == 0; // only process header rows
```
Comment (ichinaski) [inline GAE/src/org/waterforpeople/mapping/dataexport/RawDataSpreadsheetImporter.java]: I tend to be wary of assertions in non-tests, production code. Why would we want an AssertionError being thrown here? As far as I can tell, we are not handling that particular error. Also, are assertions enabled in GAE?

**Comment by ichinaski** [comment]
Just raised a small concern re: assertions (s/GAE/FLOW-services if this code is meant to be run elsewhere). Otherwise all looks good.

Edit: copyright header

File: GAE/src/org/waterforpeople/mapping/dataexport/RawDataSpreadsheetImporter.java (hunk 1 above)
Comment (muloem) [inline GAE/src/org/waterforpeople/mapping/dataexport/RawDataSpreadsheetImporter.java]: The code will run under FLOW services so no worries about GAE support.  In this case the main motivation is to ensure the method is only used for checking cells in the header row.

**Comment by ichinaski** [comment]
Code reviewed!

**Pull request merged**